### PR TITLE
gh-62897: Update PyUnicode parameter names

### PR DESCRIPTION
**Pull request by CraftSpider**
Per the linked issue, PyUnicode functions/methods had many different parameter names, this PR aims to unify them. There is still at least one unanswered question on the bug-tracker, but it seemed a good idea to get a PR of the existing fixes started to solicit active comments.

https://bugs.python.org/issue18697

* Issue: gh-62897

### CODE REVIEW

File: Doc/c-api/unicode.rst [hunk 1]
```diff
@@ -398,7 +398,7 @@ APIs:
    .. versionadded:: 3.3
 
 
-.. c:function:: PyObject* PyUnicode_FromKindAndData(int kind, const void *buffer, \
+.. c:function:: PyObject* PyUnicode_FromKindAndData(int kind, const void *str, \
```
Comment (serhiy-storchaka) [inline Doc/c-api/unicode.rst]: This is not a string of `char`s. It is a sequence of 8-, 16- or 32-bit integers, depending of the kind. *buffer* is a good name.

File: Doc/c-api/unicode.rst [hunk 2]
```diff
@@ -627,7 +627,7 @@ APIs:
    .. versionadded:: 3.3
 
 
-.. c:function:: PyObject* PyUnicode_Substring(PyObject *str, Py_ssize_t start, \
+.. c:function:: PyObject* PyUnicode_Substring(PyObject *self, Py_ssize_t start, \
```
Comment (serhiy-storchaka) [inline Doc/c-api/unicode.rst]: Why not *unicode*?

File: Doc/c-api/unicode.rst [hunk 3]
```diff
@@ -833,7 +833,7 @@ To encode and decode file names and other environment strings,
 argument parsing, the ``"O&"`` converter should be used, passing
 :c:func:`PyUnicode_FSConverter` as the conversion function:
 
-.. c:function:: int PyUnicode_FSConverter(PyObject* obj, void* result)
+.. c:function:: int PyUnicode_FSConverter(PyObject* arg, void* addr)
```
Comment (serhiy-storchaka) [inline Doc/c-api/unicode.rst]: I think *obj* is good name.

**Review by serhiy-storchaka — COMMENTED** [review]
Do not change C sources. Update only the documentation. Don't forget to update the text following the signature.

**Comment by CraftSpider** [comment]
Alright, I'll revert the code changes. Phrasing and discussion on the issue made me think it also wanted the code names unified with the doc names

File: Doc/c-api/unicode.rst (hunk 2 above)
Comment (CraftSpider) [reply Doc/c-api/unicode.rst]: As mentioned below, thought goal was unity with code files. Honestly not sure why I didn't change it anyways, oversight on my part

**Review by mangrisano — APPROVED** [review]
LGTM. Thank you.

File: Doc/c-api/unicode.rst [hunk 4]
```diff
@@ -179,7 +179,7 @@ access internal read-only data of Unicode objects:
    .. versionadded:: 3.3
 
 
-.. c:function:: Py_UCS4 PyUnicode_READ_CHAR(PyObject *o, Py_ssize_t index)
+.. c:function:: Py_UCS4 PyUnicode_READ_CHAR(PyObject *unicode, Py_ssize_t index)
 
    Read a character from a Unicode object *o*, which must be in the "canonical"
```
Comment (matrixise) [inline Doc/c-api/unicode.rst]: should be *unicode*

File: Doc/c-api/unicode.rst [hunk 5]
```diff
@@ -409,7 +409,7 @@ APIs:
    .. versionadded:: 3.3
 
 
-.. c:function:: PyObject* PyUnicode_FromStringAndSize(const char *u, Py_ssize_t size)
+.. c:function:: PyObject* PyUnicode_FromStringAndSize(const char *str, Py_ssize_t size)
 
    Create a Unicode object from the char buffer *u*.  The bytes will be
```
Comment (matrixise) [inline Doc/c-api/unicode.rst]: ditto with *u* and *str*

File: Doc/c-api/unicode.rst [hunk 6]
```diff
@@ -421,7 +421,7 @@ APIs:
    :c:func:`PyUnicode_New`.
 
 
-.. c:function:: PyObject *PyUnicode_FromString(const char *u)
+.. c:function:: PyObject *PyUnicode_FromString(const char *str)
 
    Create a Unicode object from a UTF-8 encoded null-terminated char buffer
    *u*.
```
Comment (matrixise) [inline Doc/c-api/unicode.rst]: ditto with *u* and *str*

File: Doc/c-api/unicode.rst [hunk 7]
```diff
@@ -627,7 +627,7 @@ APIs:
    .. versionadded:: 3.3
 
 
-.. c:function:: PyObject* PyUnicode_Substring(PyObject *str, Py_ssize_t start, \
+.. c:function:: PyObject* PyUnicode_Substring(PyObject *unicode, Py_ssize_t start, \
                                               Py_ssize_t end)
 
    Return a substring of *str*, from character index *start* (included) to
```
Comment (matrixise) [inline Doc/c-api/unicode.rst]: *unicode*

File: Doc/c-api/unicode.rst [hunk 8]
```diff
@@ -636,7 +636,7 @@ APIs:
    .. versionadded:: 3.3
 
 
-.. c:function:: Py_UCS4* PyUnicode_AsUCS4(PyObject *u, Py_UCS4 *buffer, \
+.. c:function:: Py_UCS4* PyUnicode_AsUCS4(PyObject *unicode, Py_UCS4 *buffer, \
                                           Py_ssize_t buflen, int copy_null)
 
    Copy the string *u* into a UCS4 buffer, including a null character, if
```
Comment (matrixise) [inline Doc/c-api/unicode.rst]: *unicode*

**Review by matrixise — CHANGES_REQUESTED** [review]
Thank you for your contribution, but could you check your PR, because you change some parameters, but the documentation is not updated with these new names.

**Comment by CraftSpider** [comment]
Not sure why there's an error about seeing `:end`, as that was in the original docs and was preserved unchanged

**Comment by csabella** [comment]
@CraftSpider, in order to move issues along once changes have been made, this repo uses bots.  Please take a look at [this message](https://github.com/python/cpython/pull/12680#issuecomment-530383012) for more info about that.  Please also resolve the merge conflict before adding that comment requesting further review.  Thanks!

**Comment by CraftSpider** [comment]
I have made the requested changes; please review again

Thank you for the patience, I got into other projects and basically forgot about this for a while

**Comment by CraftSpider** [comment]
Fixed a merge error caused by other changes since this was made, @matrixise can you please review this when you have time?

**Review by furkanonder — APPROVED** [review]

**Comment by furkanonder** [comment]
LGTM

**Comment by arhadthedev** [comment]
Could anyone help to sort out the new `c:... reference target not found: ...` warnings please?

**Pull request merged**